tiledb/sm/filesystem/s3.cc

```diff
@@ -633,13 +633,11 @@ Status S3::disconnect() {
             Aws::S3::Model::AbortMultipartUploadRequest abort_request =
                 make_multipart_abort_request(*state);
             auto outcome = client_->AbortMultipartUpload(abort_request);
-            if (!outcome.IsSuccess()) {
-              const Status st = LOG_STATUS(Status_S3Error(
-                  std::string("Failed to disconnect and flush S3 objects. ") +
-                  outcome_error_message(outcome)));
-              if (!st.ok()) {
-                ret_st = st;
-              }
+            const Status st = LOG_STATUS(Status_S3Error(
+                std::string("Failed to disconnect and flush S3 objects. ") +
+                outcome_error_message(outcome)));
+            if (!st.ok()) {
```

Comment:
This check was already in place before you made your change, but I think `if (!st.ok())` will always be not-ok since we construct the status with an error just above. The `if` statement could be removed and we can probably assign `ret_st` directly.

Reply:
Is there a case where the `AbortMultipartUpload` would actually fail? (maybe temporary loss of network connection?) I'm wondering if instead of removing the `if(!outcome.IsSuccess())` we could do something like this so that error isn't lost.

```C++
if (!outcome.IsSuccess()) {
  ret_st = LOG_STATUS(Status_S3Error(
      std::string("Failed to disconnect and flush S3 objects. ") +
      outcome_error_message(outcome)));
} else {
  ret_st = LOG_STATUS(Status_S3Error(
      std::string("Failed to disconnect and flush S3 objects. ")));
}
```

Reply:
Done!